pkg/shared/contextutil/util.go

```diff
@@ -93,6 +97,37 @@ func GetCurrentRegistryInstance(f *factory.Factory) (*registrymgmtclient.Registr
 
 }
 
+func GetCurrentKafkaInstanceDataPlane(f *factory.Factory) (*kafkainstanceclient.APIClient, *kafkamgmtclient.KafkaRequest, error) {
+
+	svcContext, err := f.ServiceContext.Load()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	conn, err := f.Connection()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currCtx, err := GetCurrentContext(svcContext, f.Localizer)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	fmt.Println("inside get kafka dp")
+
+	api, kafkaInstance, err := conn.API().KafkaAdmin(currCtx.KafkaID)
+
+	fmt.Println("error - ", err)
+	if kafkamgmtv1errors.IsAPIError(err, kafkamgmtv1errors.ERROR_7) {
```

Comment:
Problem with this approach is, there is no way to identify a 404 error using error codes, which makes it difficult to map an error to a custom error message like we have for `GetCurrentKafkaInstance`. The only way to go seems to be using a `String.contains`, which is not a preferred way for error handling.

Reply:
Cc @jackdelahunt @dimakis 

Need a second opinion if we should follow the `string.contains` hack to optimize, or just let the data plane commands as they are (making two calls that return the current Kafka instance).

Reply:
> there is no way to identify a 404 error using error codes

I was not aware of this can you explain why this is?